src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala

```diff
@@ -245,87 +133,264 @@ class ScalaPageCreator(
 
-        def modifiedContent = content(0).copy(
-            (content(0).getChildren.asScala ++ List(addedContent)).asJava,
-            content(0).getDci,
-            content(0).getSourceSets,
-            content(0).getStyle,
-            content(0).getExtra
-        )
-        if !supertypes.isEmpty then modifyContentGroup(content, modifiedContent) else defContent
-    }
+    extension (b: ScalaPageContentBuilder#ScalaDocumentableContentBuilder):
+        def descriptionIfNotEmpty(d: Documentable): ScalaPageContentBuilder#ScalaDocumentableContentBuilder = {
+            val desc = contentForDescription(d).asScala.toSeq
+            val res = if desc.isEmpty then b else b
+                .sourceSetDependentHint(
+                    Set(d.getDri), 
+                    d.getSourceSets.asScala.toSet, 
+                    kind = ContentKind.SourceSetDependentHint, 
+                    styles = Set(TextStyle.UnderCoverText)
+                ) { sourceSetBuilder => sourceSetBuilder
+                        .addChildren(desc)
+                }
+            res
+        }
 
-    def insertKnownSubclasses(clazz: DClass, defContent: ContentGroup): ContentGroup = {
-        val content = getContentGroupWithParents(defContent, p => p.getStyle.asScala.contains(ContentStyle.TabbedContent))
-        val subtypes = clazz.get(InheritanceInfo).knownChildren
+        def contentForComments(d: Documentable) = b
 
-        def contentForType(
-            bdr: ScalaPageContentBuilder#ScalaDocumentableContentBuilder,
-            b: DRI
-        ): ScalaPageContentBuilder#ScalaDocumentableContentBuilder = bdr
-            .driLink(b.getClassNames, b)
-            
+        def contentForDescription(d: Documentable) = {
+            val specialTags = Set[Class[_]](classOf[Description])
+            val tags = d.getDocumentation.asScala.toList.flatMap( (pd, doc) => doc.getChildren.asScala.map(pd -> _).toList )
 
-        def buildBlock: ScalaPageContentBuilder#ScalaDocumentableContentBuilder => ScalaPageContentBuilder#ScalaDocumentableContentBuilder = bdr => bdr
-            .header(2, "Known subtypes")()
-            .group(
-                styles = Set(ContentStyle.WithExtraAttributes), 
-                extra = PropertyContainer.Companion.empty plus SimpleAttr.Companion.header("Known subtypes")
-            ){ gbdr => gbdr
-                .group(kind = ContentKind.Symbol, styles = Set(TextStyle.Monospace)){ grbdr => grbdr
-                    .list(subtypes)(contentForType)
-                }
-            }
+            val platforms = d.getSourceSets.asScala.toSet
 
-        def addedContent = ScalaPageContentBuilder(commentsToContentConverter, signatureProvider, logger).contentForDocumentable(
-            clazz, 
-            buildBlock = buildBlock
-        )
+            val description = tags.collect{ case (pd, d: Description) => (pd, d) }.groupBy(_(0)).map( (key, value) => key -> value.map(_(1)))
```

Comment:
I'm not 100% sure that this is the correct place, but I guess that the first `Description` tag here should be dropped. The first tag was previously treated as the short description, and wasn't displayed on the class pages. Now it is, which results in pages looking like https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-b/index.html and https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-a/index.html (note how paragraphs are duplicated).

Reply:
Done